test/specs/productDetails.js

```diff
@@ -1,29 +1,199 @@
+            );
+        })
+    })
+
+    describe("Burger Menu Tests", () => {
+
+        beforeEach(async function () {
+            await LoginPage.open();
+            await LoginPage.login(CREDENTIALS.standard, CREDENTIALS.password);
             await ProductsPage.allItemsImages[0].click()
+        });
+
+        it("check Burger Menu UI elements", async () => {
+            const bmItemsText = await ProductsPage.burgerMenuItemsText()
+
+            expect(bmItemsText).toContain(BurgerMenuText.allItems.toUpperCase());
+            expect(bmItemsText).toContain(BurgerMenuText.about.toUpperCase());
+            expect(bmItemsText).toContain(BurgerMenuText.logout.toUpperCase());
+            expect(bmItemsText).toContain(BurgerMenuText.resetAppState.toUpperCase());
+
+            await ProductsPage.btnLogout.click();
+        })
+
+        it("the 'X' button should close the burger menu", async () => {
+            await ProductsPage.btnBurgerMenu.click();
+            await ProductsPage.btnCloseBurgerMenu.waitForClickable();
+            await ProductsPage.btnCloseBurgerMenu.click();
+            await ProductsPage.btnBurgerMenu.waitForClickable();
+
+            expect(await ProductsPage.btnCloseBurgerMenu.isDisplayed()).toBe(false)
+            expect(await ProductsPage.btnLogout.isDisplayed()).toBe(false)
+            expect(await ProductsPage.btnBurgerMenu.isDisplayed()).toBe(true)
+
+            await LoginPage.logout();
+        })
+
+        it("the 'ALL ITEMS' button should redirect to the Products page", async () => {
+            await ProductsPage.btnBurgerMenu.click();
+            await ProductsPage.btnAllItems.waitForClickable();
+            await ProductsPage.btnAllItems.click();
 
-            expect(await ProductDetailedPage.detailsImage.isDisplayed()).toBe(true);
-            expect(await ProductDetailedPage.detailsImage.getAttribute('src')).toEqual(ImageSource.backpack);
+            expect(await browser.getUrl()).toContain(ProductsPage.url);
+        })
+
+        it("the 'About' button should redirect to the 'https://saucelabs.com/' url", async () => {
+            await LoginPage.open();
+            await LoginPage.login(CREDENTIALS.standard, CREDENTIALS.password);
+            await ProductsPage.btnBurgerMenu.click();
+            await ProductsPage.btnAbout.waitForClickable();
+            await ProductsPage.btnAbout.click();
+
+            expect(await browser.getUrl()).toEqual("https://saucelabs.com/");
+        })
+
+        it("the 'Logout' button should redirect to the Login page", async () => {
+            await LoginPage.open();
+            await LoginPage.login(CREDENTIALS.standard, CREDENTIALS.password);
+            await ProductsPage.btnBurgerMenu.click();
+            await ProductsPage.btnLogout.waitForClickable();
+            await ProductsPage.btnLogout.click();
+
+            expect(await browser.getUrl()).toContain(LoginPage.url);
+        })
+    })
```

Comment:
your already wrote the tests for burger menu. better create a separate file for burger menu functionality tests, and in productDetails.js and products.js check only that the burger menu button is displayed.

Reply:
the file BurgerMenu has been added into the specs folder and the modifications have been made in products and productDetails